### PR TITLE
Update the IMS pre-processing time and IMS snow data format

### PR DESCRIPTION
**Pull request by jiaruidong2017**
# Description
Since the IMS snow products was ingested at about 23:00 UTC, and the IMS snow cover data is in the ascii format. Therefore, we propose to assimilate the IMS snow cover data in the ascii format at 00z cycle. 

This PR updates the IMS snow data pre-processing time from being the 18z to the 00z cycle, and also changes to assimilate the IMS snow data in ascii format from the operational dump. 

An IMS snow data in ascii format will be staged to global dump directory for the CI test `C96_atmaerosnowDA.yaml` at: 
/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/GlobalWorkflow/para_gfs/glopara_dump/gdas.20211221/00/atmos/gdas.t00z.imssnow96.asc

This PR depends on https://github.com/NOAA-EMC/GDASApp/pull/1228

# Type of change

- Bug fix (fixes something broken)
- New feature (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
Cycled test on Hera

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — DISMISSED** [review]
Conditionally approve pending CI

File: scripts/exglobal_prep_snow_obs.py [hunk 1]
```diff
@@ -21,5 +21,6 @@
     # Instantiate the snow prepare task
     SnowAnl = SnowAnalysis(config)
     SnowAnl.prepare_GTS()
-    if f"{ SnowAnl.task_config.cyc }" == '18':
+    SnowAnl.task_config.cyc = f'{SnowAnl.task_config.cyc:02d}'
```
Comment (CoryMartin-NOAA) [inline scripts/exglobal_prep_snow_obs.py]: Why is this needed?

Comment (jiaruidong2017) [reply scripts/exglobal_prep_snow_obs.py]: To keep the leading '0' for the 00z cycle to be '00', so that the following `if f"{ SnowAnl.task_config.cyc }" == '00':` is true for the 00z

Comment (CoryMartin-NOAA) [reply scripts/exglobal_prep_snow_obs.py]: can't it just be `if SnowAnl.task_config.cyc == 0`?

Comment (jiaruidong2017) [reply scripts/exglobal_prep_snow_obs.py]: Made the according changes. Thanks.

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Comment by CoryMartin-NOAA** [comment]
Looks good, the g-w team will need to work with @jiaruidong2017 to update the test data before kicking off CI

**Comment by aerorahul** [comment]
@KateFriedman-NOAA @WalterKolczynski-NOAA 
Test data needs to be updated as part of this PR.  
Please coordinate w/ @jiaruidong2017

**Comment by jiaruidong2017** [comment]
@KateFriedman-NOAA @WalterKolczynski-NOAA Would you please copy the test data: 
`/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/GlobalWorkflow/para_gfs/glopara_dump/gdas.20211221/00/atmos/gdas.t00z.imssnow96.asc` on hera to the global dump for CI test. Thanks.

**Comment by KateFriedman-NOAA** [comment]
@jiaruidong2017 I have copied the file into the GDA on Hera:
```
[role.glopara@hfe11 atmos]$ pwd
/scratch1/NCEPDEV/global/glopara/dump/gdas.20211221/00/atmos
[role.glopara@hfe11 atmos]$ ll *ims*
-rw-r--r-- 1 role.glopara global 37756368 Jul 17 19:22 gdas.t00z.imssnow96.asc
-rw-r--r-- 1 role.glopara global 10259780 Dec 21  2021 gdas.t00z.imssnow96.grib2
```
I have also synced this to the GDA on both WCOSS2s. @WalterKolczynski-NOAA please sync this file to the GDA on Orion, thanks!

**Comment by RussTreadon-NOAA** [comment]
We also need to copy 
`/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/GlobalWorkflow/para_gfs/glopara_dump/gfs.20211221/00/atmos/gfs.t00z.imssnow96.asc`

to 

`/scratch1/NCEPDEV/global/glopara/dump/gfs.20211221/00/atmos`

The 20211221 00Z gfsprepsnowobs from g-w C96_atmaerosnowDA CI fails because `gfs.t00z.imssnow96.asc` can not be found.

**Comment by jiaruidong2017** [comment]
> We also need to copy `/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/GlobalWorkflow/para_gfs/glopara_dump/gfs.20211221/00/atmos/gfs.t00z.imssnow96.asc`
> 
> to
> 
> `/scratch1/NCEPDEV/global/glopara/dump/gfs.20211221/00/atmos`
> 
> The 20211221 00Z gfsprepsnowobs from g-w C96_atmaerosnowDA CI fails because `gfs.t00z.imssnow96.asc` can not be found.

Thanks @RussTreadon-NOAA. Yes, @KateFriedman-NOAA would you please make the copy. Thanks.

**Comment by CoryMartin-NOAA** [comment]
I think this PR is needed to be merged to fix failures seen in CI testing on #2749

**Comment by guillaumevernieres** [comment]
Could you resolve the conflict @jiaruidong2017 ?

**Comment by jiaruidong2017** [comment]
> Could you resolve the conflict @jiaruidong2017 ?

@guillaumevernieres I learn to try it by merging branch `develop`. Is this correct way to do it?

**Comment by KateFriedman-NOAA** [comment]
@jiaruidong2017 @RussTreadon-NOAA I have also copied the gfs file into the GDA on Hera:

```
[role.glopara@hfe11 atmos]$ pwd
/scratch1/NCEPDEV/global/glopara/dump/gfs.20211221/00/atmos
[role.glopara@hfe11 atmos]$ ll *ims*
-rw-r--r-- 1 role.glopara global 37756368 Jul 17 23:15 gfs.t00z.imssnow96.asc
-rw-r--r-- 1 role.glopara global 10259780 Dec 21  2021 gfs.t00z.imssnow96.grib2
```
...and synced it to both WCOSS2s. @WalterKolczynski-NOAA please also sync this file to Orion, thanks!

**Comment by RussTreadon-NOAA** [comment]
Thank you @KateFriedman-NOAA for updating the GDA.  I reran the failed 20211221 00Z gfsprepsnowobs from C96_atmaerosnowDA CI.  The job successfully ran to completion.

**Comment by WalterKolczynski-NOAA** [comment]
Synchronized to MSU.

**Comment by CoryMartin-NOAA** [comment]
@guillaumevernieres any idea what happened here? Did something change in SOCA with MOM6?

**Comment by RussTreadon-NOAA** [comment]
@jiaruidong2017 , did you intend to update the `sorc/gdas.cd` hash when you merged g-w `develop` into `jiaruidong2017:feature/ims_ascii` at [d10c3f5](https://github.com/NOAA-EMC/global-workflow/pull/2779/commits/d10c3f59fd648a6b8a0d4710d73c842e0feca172)?

**Comment by jiaruidong2017** [comment]
@RussTreadon-NOAA 

@guillaumevernieres is working on this to merge my changes to his branch.

**Comment by RussTreadon-NOAA** [comment]
Thank you @jiaruidong2017 for the update.  

Thank you @guillaumevernieres for merging PR #2779 into #2749.

**Comment by DavidHuber-NOAA** [comment]
Manually killed CI tests on Hera and Hercules which had stalled.  Will restart testing on #2749 when merged and approved.

**Comment by jiaruidong2017** [comment]
This PR will be merged into the PR #2749.
